src/main/java/ch/kalunight/zoe/repositories/InfoChannelRepository.java

```diff
@@ -0,0 +1,193 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ch.kalunight.zoe.model.dto.DTO;
+
+public class InfoChannelRepository {
+
+  private static final String SELECT_INFOCHANNEL_WITH_GUILD_ID = "SELECT " + 
+      "info_channel.infochannel_id, " + 
+      "info_channel.infochannel_fk_server, " + 
+      "info_channel.infochannel_channelid " + 
+      "FROM info_channel " + 
+      "INNER JOIN server ON info_channel.infochannel_fk_server = server.serv_id " +
+      "WHERE serv_guildId = %d";
+
+  private static final String SELECT_INFO_PANEL_WITH_GUILDID = "SELECT " + 
+      "info_panel_message.infopanel_id, " + 
+      "info_panel_message.infopanel_fk_infochannel, " + 
+      "info_panel_message.infopanel_messageid " + 
+      "FROM server " + 
+      "INNER JOIN info_channel ON server.serv_id = info_channel.infochannel_fk_server " + 
+      "INNER JOIN info_panel_message ON info_channel.infochannel_id = info_panel_message.infopanel_fk_infochannel " + 
+      "WHERE server.serv_guildid = %d";
+
+  private static final String INSERT_INTO_INFOCHANNEL = "INSERT INTO info_channel (infochannel_fk_server, infochannel_channelid) " +
+      "VALUES (%d, %d)";
+
+  private static final String INSERT_INTO_INFO_PANEL_MESSAGE = "INSERT INTO info_panel_message " +
+      "(infopanel_fk_infochannel, infopanel_messageid) VALUES (%d, %d)";
+
+  private static final String UPDATE_INFOCHANNEL_WITH_GUILD_ID =
+      "UPDATE info_channel " +
+          "SET infochannel_channelid = %d " +
+          "FROM server " +
+          "WHERE server.serv_guildId = %d AND " +
+          "server.serv_id = info_channel.infochannel_fk_server";
+
+  private static final String DELETE_INFOCHANNEL_WITH_GUILD_ID = "DELETE FROM info_channel " +
+      "USING server " +
+      "WHERE server.serv_id = info_channel.infochannel_fk_server AND server.serv_guildid = %d";
+
+  private static final String DELETE_INFO_PANEL_MESSAGE_ID = 
+      "DELETE FROM info_panel_message WHERE infopanel_id = %d";
+
+  private static final Logger logger = LoggerFactory.getLogger(InfoChannelRepository.class);
+
+  private InfoChannelRepository() {
+    //Hide default public constructor
+  }
+
+  @Nullable
+  public static DTO.InfoChannel getInfoChannel(long guildId) throws SQLException{
+    ResultSet result = null;
+    try (Connection conn = RepoRessources.getConnection();
+        Statement query = conn.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);) {
+
+      String finalQuery = String.format(SELECT_INFOCHANNEL_WITH_GUILD_ID, guildId);
+      result = query.executeQuery(finalQuery);
+      int rowCount = result.last() ? result.getRow() : 0;
+      if(rowCount == 0) {
+        return null;
+      }
+      return new DTO.InfoChannel(result);
+    }finally {
+      RepoRessources.closeResultSet(result);
+    }
+  }
+
+  public static List<DTO.InfoPanelMessage> getInfoPanelMessages(long guildId) throws SQLException {
+    ResultSet result = null;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=4739924)